src/components/Hoverable/ActiveHoverable.tsx

```diff
@@ -0,0 +1,143 @@
+import {cloneElement, forwardRef, MutableRefObject, Ref, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import {DeviceEventEmitter} from 'react-native';
+import CONST from '@src/CONST';
+import HoverableProps from './types';
+
+/**
+ * Assigns a ref to an element, either by setting the current property of the ref object or by calling the ref function
+ * @param ref The ref object or function.
+ * @param element The element to assign the ref to.
+ */
+function assignRef(ref: ((instance: HTMLElement | null) => void) | MutableRefObject<HTMLElement | null>, element: HTMLElement) {
+    if (!ref) {
+        return;
+    }
+    if (typeof ref === 'function') {
+        ref(element);
+    } else if ('current' in ref) {
+        // eslint-disable-next-line no-param-reassign
+        ref.current = element;
+    }
+}
+
+type ActiveHoverableProps = Omit<HoverableProps, 'disabled'>;
+
+function ActiveHoverable({onHoverIn, onHoverOut, shouldHandleScroll, children}: ActiveHoverableProps, outerRef: Ref<HTMLElement>) {
+    const [isHovered, setIsHovered] = useState(false);
+
+    const ref = useRef<HTMLElement | null>(null);
+    const isScrolling = useRef(false);
+    const isHoveredRef = useRef(false);
+
+    const updateIsHovered = useCallback(
+        (hovered: boolean) => {
+            isHoveredRef.current = hovered;
+            if (shouldHandleScroll && isScrolling.current) {
+                return;
+            }
+            setIsHovered(hovered);
+        },
+        [shouldHandleScroll],
+    );
+
+    // Expose inner ref to parent through outerRef. This enable us to use ref both in parent and child.
+    useImperativeHandle<HTMLElement | null, HTMLElement | null>(outerRef, () => ref.current, []);
```

Comment:
I'm unsure about this line... Is it guaranteed that `ref` will never change once initially set? In other words, is it guaranteed that only one HTML element will correspond to this component?

Reply:
According to the [docs](https://react.dev/reference/react/useImperativeHandle#useimperativehandle), it should work like that. We've tested this solution and there were no issues reported.

What the hook does in this case, is it *hooks* (pun intended) into ref in order to make it available in the `Hoverable`, while also forwarding it to the parent. 

I have not found a better solution and AFAIK is a state of the art for such mechanism. Let me know if you know anything better 👍